tests/test_waku_netconfig.nim

```diff
@@ -0,0 +1,351 @@
+{.used.}
+
+import
+  chronos,
+  libp2p/multiaddress,
+  testutils/unittests
+
+import
+    ../../waku/node/config,
+    ../../waku/waku_enr/capabilities,
+    ../../apps/wakunode2/external_config
+
+
+proc defaultTestWakuFlags(): CapabilitiesBitfield =
+  CapabilitiesBitfield.init(
+      lightpush = false,
+      filter = false,
+      store = false,
+      relay = true
+  )
+    
+suite "Waku NetConfig":
+  
+  asyncTest "Create NetConfig with default values":
+
+    let conf = defaultTestWakuNodeConf()
+    
+    let wakuFlags = defaultTestWakuFlags()
+    
+    let netConfigRes = NetConfig.init(
+      bindIp = conf.listenAddress,
+      bindPort = conf.tcpPort,
+      extIp = none(ValidIpAddress),
+      extPort = none(Port),
+      extMultiAddrs = @[],
+      wsBindPort = conf.websocketPort,
+      wsEnabled = conf.websocketSupport,
+      wssEnabled = conf.websocketSecureSupport,
+      dns4DomainName = none(string),
+      discv5UdpPort = none(Port),
+      wakuFlags = some(wakuFlags)
+    )
+
+    check:
+      netConfigRes.isOk()
+
+asyncTest "AnnouncedAddresses contains only bind address when no external addresses are provided":
+
+    let conf = defaultTestWakuNodeConf()
+        
+    let netConfigRes = NetConfig.init(
+      bindIp = conf.listenAddress,
+      bindPort = conf.tcpPort
+    )
+     
+    require:
+      netConfigRes.isOk()
+
+    let netConfig = netConfigRes.get()
+
+    check:
+      netConfig.announcedAddresses.len == 1  # Only bind address should be present
+      netConfig.announcedAddresses[0] == formatListenAddress(ip4TcpEndPoint(conf.listenAddress, conf.tcpPort))
+
+
+asyncTest "AnnouncedAddresses contains external address if extIp/Port are provided":
```

Comment:
Ditto in terms of indentation and for other tests (I think they should be single indented to be considered part of "Waku Netconfig" `suite`

Reply:
Good catch! For some reason the tests were still being run as part of the test suite with that indentation, which is pretty interesting. Just pushed a fix for it - thank you!